### PR TITLE
Remove CircularArray (NFC)

### PR DESCRIPTION
**Pull request by keithw**
This PR removes the CircularArray class (only used in one place) and its unit test, in favor of a small replacement inside the WastParser.

I suspect we could also just use a std::queue with probably only a small performance penalty -- maybe that's cleaner/simpler?

### CODE REVIEW

File: include/wabt/wast-parser.h [hunk 1]
```diff
@@ -262,7 +262,28 @@ class WastParser {
   Errors* errors_;
   WastParseOptions* options_;
 
-  CircularArray<Token, 2> tokens_;
+  // two-element queue of upcoming tokens
```
Comment (sbc100) [inline include/wabt/wast-parser.h:265]: I'd still rather give this data type a name and define it out-of-line, if only to make the containing class easier to read.

File: include/wabt/wast-parser.h [hunk 2]
```diff
@@ -262,7 +262,28 @@ class WastParser {
...
+    }
+    void pop_front() {
+      assert(tokens[i]);
+      tokens[i].reset();
+      i = !i;
+    }
+    const Token& at(bool n) const { return tokens[i ^ n].value(); }
+    const Token& front() const { return at(0); }
+    bool empty() const { return !tokens[i]; }
+    size_t size() const { return empty() ? 0 : 1 + tokens[!i].has_value(); }
```
Comment (sbc100) [inline include/wabt/wast-parser.h:285]: Are all these methods actually used in this file?

**Review by sbc100 — APPROVED** [review]

File: include/wabt/wast-parser.h (hunk 1 above)
Comment (sbc100) [reply include/wabt/wast-parser.h:265]: Maybe `TokenQueue`?

Comment (keithw) [reply include/wabt/wast-parser.h:265]: Done.

File: include/wabt/wast-parser.h (hunk 2 above)
Comment (keithw) [reply include/wabt/wast-parser.h:285]: No... they're just used in wast-parser.cc. Moved the definitions to the .cc file.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**